ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -373,6 +383,54 @@ void fillChunks() {
   }
 
   /**
+   * Submits encrypt job for the given {@link PutChunk} and processes the callback for the same
+   * @param putChunk the {@link PutChunk} that needs to be encrypted
+   * @param encryptContent {@code true} if content needs to be encrypted. {@code false} otherwise
+   */
+  private void encryptChunk(PutChunk putChunk, boolean encryptContent) {
+    try {
+      logger.trace("Chunk at index {} moves to {} state", putChunk.chunkIndex, ChunkState.Encrypting);
+      putChunk.state = ChunkState.Encrypting;
+      putChunk.chunkEncryptReadyAtMs = time.milliseconds();
+      logger.trace("Submitting encrypt job for chunk at index {}", putChunk.chunkIndex);
+      cryptoJobHandler.submitJob(
+          new EncryptJob(passedInBlobProperties.getAccountId(), passedInBlobProperties.getContainerId(),
+              encryptContent ? putChunk.buf : null, ByteBuffer.wrap(putChunk.chunkUserMetadata), kms.getRandomKey(),
+              cryptoService, kms, (EncryptJob.EncryptJobResult result, Exception exception) -> {
+            logger.trace("Processing encrypt job callback for chunk at index " + putChunk.chunkIndex);
+            if (exception == null && !isOperationComplete()) {
+              if (encryptContent) {
+                putChunk.buf = result.getEncryptedBlobContent();
+              }
+              putChunk.encryptedPerBlobKey = result.getEncryptedKey();
+              putChunk.chunkUserMetadata = result.getEncryptedUserMetadata().array();
+              logger.trace("Completing encrypt job result for chunk at index " + putChunk.chunkIndex);
+              putChunk.prepareForSending();
+              putChunk.chunkReadyAtMs = time.milliseconds();
+            } else {
+              if (getOperationException() == null) {
+                logger.trace("Setting exception {} from encrypt of chunk at index {} ", exception, putChunk.chunkIndex);
+                setOperationExceptionAndComplete(new RouterException(
+                    "Exception thrown on encrypting the content for chunk at index " + putChunk.chunkIndex, exception,
+                    RouterErrorCode.UnexpectedInternalError));
+              } else {
+                logger.trace(
+                    "Ignoring exception {} from encrypt job for chunk at index {} as operation exception {} is set already",
```

Comment:
same here

Reply:
also should this be at the `error` level?

Reply:
I saw that no other exceptions in the same file log at error level (all of them are in trace). So, I am assuming that frontend will take care of logging any error msgs.